### PR TITLE
Fixup documentation for `swap_sub_trees`

### LINKED ISSUE #3: Control over children ordering
I am wondering if any guarantees on the order of children would be a goal or non-goal of this library?

It seems `insert_with_parent` currently only pushes to the end of the children list.
If I would like to replace a child and keep the current position, this does not seem to be easily possible and I don't even know, if I can rely on `insert_with_parent` always appending at the end, because this is not documented.

For my use case I just have a left and a right child and it would be great to be able to differentiate between those. But I have no good idea, how to provide this functionality in the context of this library with an arbitrary amount of child nodes.

Possible API suggestions:
- Split insert into two functions push at the front and back of the children respectively.
    - Not a very nice API in my opinion. The use case is not very obvious at first.
    - Actual sorting would be difficult to emulate given only these functions
    - Enough for my use case
    - Easy to implement
- Allow to sorting of children via a closure
    - Makes a good API
    - Can be used to reorder children, after "replacing" (remove/insert), but needs to track old positions manually
    - That means not very nice for my use case
    - Easy to implement (https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.sort_by_key)
- Allow swapping of children
    - Also not a very nice API. The use case is not very obvious at first.
    - Actual sorting would be difficult to emulate given only these functions
    - Enough for my use case
    - Easy to implement
- Allow replacement of `Node`s and swapping of `Node`s.
    - Swapping could take two `NodeId`s
    - Replacing could insert a new node, removing the old in the process.
    - Still does not make very strong order guarantees except to keep the old order
    - Feels like it has use cases beyond this

I mostly opened this issue to discuss, if you would want to support any order guarantees anyway. The rest is just a little bit of brainstorming, please feel free to edit/extend/ignore it, however you like.

### PR DESCRIPTION
**Pull request by Drakulix**
Fixes #3

### CODE REVIEW

File: src/tree.rs [hunk 1]
```diff
@@ -629,11 +629,14 @@ impl<T> Tree<T> {
     ///
     /// Swaps two `Node`s including their children given their `NodeId`s.
     ///
-    /// Returns an empty `Result` containing a `NodeIdError` if one occurred on either provided `Id`.
+    /// If one `Node` is a descendant of the other getting swapped, the former *upper* `Node` is attached as last child of the former *lower* `Node` after the swap. (The *lower* is taking the *uppers* original position as usual.)
+    /// The subtree of the former *upper* node is not touched except for the *lower* node moved including all its children.
+    ///
```
Comment (iwburns) [inline src/tree.rs]: Could you change this:
"...is not touched except for the *lower* node moved including..."
to:
"...is not touched except that the *lower* node is moved including..."

File: src/tree.rs [hunk 2]
```diff
@@ -629,11 +629,14 @@ impl<T> Tree<T> {
...
+    /// Both `NodeId`s are still valid after this process and are not swapped.
     ///
-    /// Notes:
-    ///     - Both `NodeId`s are still valid after this process and are not swapped.
-    ///     - Keeps the positions of the `Node`s in their parents children collection.
+    /// This keeps the positions of the `Node`s in their parents children collection.
```
Comment (iwburns) [inline src/tree.rs]: Could you change this:
"...in their parents..."
to:
"...in their parents'..."
(single quote after the s)

**Comment by Drakulix** [comment]
I hope I did not miss anything

**Comment by iwburns** [comment]
Looks good to me.

You got anything else before I merge and push to crates? 😆 😄

**Comment by Drakulix** [comment]
Not this time ;)

**Comment by iwburns** [comment]
alright, it's up on crates.io

**Pull request merged**